### ISSUE TITLE
Allow for reading `null` fields when reading simple objects

### ISSUE THREAD

**Issue by jbartok** [FasterXML/jackson-jr#73]
Let's say I use `JSON.std.beanFrom(MyBean.class, INPUT)` as stated in the docs and `MyBean` has a `Date` field. If the JSON being parsed has a numeric value for that data field, then everything works ok, but if the value is `null` (for example `"birthday":null`), then I end up with: `com.fasterxml.jackson.jr.ob.JSONObjectException: Can not get long numeric value from JSON (to construct java.util.Date) from 'null'`.

This seems to be an arbitrary limitation caused by `SimpleValueReader.read` having following block:
```
case SER_DATE:
    return new Date(_fetchLong(p));
```

```
protected long _fetchLong(JsonParser p) throws IOException
    {
        JsonToken t = p.currentToken();
        if (t == JsonToken.VALUE_NUMBER_INT) {
            return p.getLongValue();
        }
        throw JSONObjectException.from(p, "Can not get long numeric value from JSON (to construct "
                +_valueType.getName()+") from "+_tokenDesc(p, t));
    }
```

Any reason to not improve on this, to return an actual `null` value instead of throwing an exception? Conceptually at least, seam the right thing to do, especially since reading into a `Map` (as opposed to POJO) works like that.

**Comment by cowtowncoder** [comment]
Just to make sure I understand: if POJO was like

```java
public class Bean {
  public Date date;
}
```

and JSON

```json
{ "date" : null }
```

then I think it absolutely should work as expected. Similarly, `Long` properties should accept `null`, and the only (?) open case would be if `long` (and other primitive values) should accept it.

It this the problem you have?

**Comment by cowtowncoder** [comment]
I can reproduce the problem as described.

**Comment by cowtowncoder** [comment]
Fixed this specific problem for 2.10.5/2.11.2 (as well as related scalar types I saw).

**Comment by jbartok** [comment]
Thank you, looking ahead for those versions to be released.
